tools/src/test/java/org/apache/kafka/tools/TransactionsCommandTest.java

```diff
@@ -437,6 +442,536 @@ public void testOldBrokerAbortTransactionWithUnknownCoordinatorEpoch(int coordin
         assertNormalExit();
     }
 
+    @Test
+    public void testFindHangingRequiresEitherBrokerIdOrTopic() throws Exception {
+        assertCommandFailure(new String[]{
+            "--bootstrap-server",
+            "localhost:9092",
+            "find-hanging"
+        });
+    }
+
+    @Test
+    public void testFindHangingRequiresTopicIfPartitionIsSpecified() throws Exception {
+        assertCommandFailure(new String[]{
+            "--bootstrap-server",
+            "localhost:9092",
+            "find-hanging",
+            "--broker-id",
+            "0",
+            "--partition",
+            "5"
+        });
+    }
+
+    private void expectListTransactions(
+        Map<Integer, Collection<TransactionListing>> listingsByBroker
+    ) {
+        expectListTransactions(null, listingsByBroker);
```

Comment:
nit: We could instantiate a new `ListTransactionsOptions` here instead of using `null`. This would remove the `null` check below.

Reply:
Yeah, I was trying to deal with the fact that `TransactionsCommand` invokes both variants of `listTransactions`. I dealt with it by modifying the no-arg call to provide the options explicitly.